tests/conftest.py

```diff
@@ -1,17 +1,322 @@
-import logging
 import pytest
+import json
+import logging
+import os
+import time
+import yaml
+import copy
 
 from ocsci import config
 from utility.environment_check import environment_checker  # noqa: F401
+from oc.openshift_ops import OCP
+from ocs import constants, ocp, defaults
+from ocs.exceptions import CommandFailed, CephHealthException
+from ocs.utils import create_oc_resource, apply_oc_resource
+from utility import templating
+from utility.aws import AWS
+from utility.retry import retry
+from utility.utils import destroy_cluster, run_cmd, get_openshift_installer, get_openshift_client, is_cluster_running
+from ocs.parallel import parallel
+from resources.ocs import OCS
+from tests import helpers
 
-
-logger = logging.getLogger(__name__)
+log = logging.getLogger(__name__)
 
 
-@pytest.fixture(scope="session", autouse=True)
 def polarion_testsuite_properties(record_testsuite_property):
     """
     Configures polarion testsuite properties for junit xml
     """
     polarion_project_id = config.REPORTING['polarion']['project_id']
     record_testsuite_property('polarion-projct-id', polarion_project_id)
+
+
+def cluster_teardown():
+    log.info("Destroying the test cluster")
+    destroy_cluster(config.ENV_DATA['cluster_path'])
+    log.info("Destroying the test cluster complete")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def cluster(request):
+    log.info("Running OCS basic installation")
+    cluster_path = config.ENV_DATA['cluster_path']
+    deploy = config.RUN['cli_params']['deploy']
+    teardown = config.RUN['cli_params']['teardown']
+    # Add a finalizer to teardown the cluster after test execution is finished
+    if teardown:
+        request.addfinalizer(cluster_teardown)
+        log.info("Will teardown cluster because --teardown was provided")
+    # Test cluster access and if exist just skip the deployment.
+    if is_cluster_running(cluster_path):
+        log.info("The installation is skipped because the cluster is running")
+        return
+    elif teardown and not deploy:
+        log.info("Attempting teardown of non-accessible cluster: %s", cluster_path)
+        return
+    elif not deploy and not teardown:
+        msg = "The given cluster can not be connected to: {}. ".format(cluster_path)
+        msg += "Provide a valid --cluster-path or use --deploy to deploy a new cluster"
+        pytest.fail(msg)
+    else:
```

Comment:
From @zmc 's comment above:
> Thought we decided not to do at deploy at all if the cluster path already existed, to prevent leaking resources?

I thought this as well and I think that check needs to happen here, correct? Theoretically we could provide a `--cluster-path` that has information about an existing cluster, however that cluster is inaccessible for some reason. We would then proceed to spin up a new cluster and place the information about that cluster on top of the files that already exist for the provided cluster. I think we need an additional check here.

```suggestion
    elif cluster_dir_is_empty:
```
Something along this line really where we check to make sure that there aren't resources for another cluster in the provided path. If there are we should fail and log an error to the user. If not we can proceed with the cluster deployment and use the provided `--cluster-path`

Reply:
spin up of cluster will only happen if --deploy is also provided, so this is same behaviour as current (if user provides -m deployment , a new deployment is expected)

Reply:
Resolving as we've decided to take this up in a separate issue/PR